VikingQuiz.Api/Controllers/QuizzesController.cs

```diff
@@ -70,121 +71,172 @@ public class QuizzesController : Controller
             }
 
-            string fileUrl;
-            AzureBlobService blobService;
-            try {
-                blobService = new AzureBlobService();
-                await blobService.InitializeBlob();
-                var previousQuizState = quizRepository.GetQuizById(id);
-
-                blobService.DeletePhoto(previousQuizState.PictureUrl);
-                fileUrl = await blobService.UploadPhoto(quizBodyData.Files[0]);
-            }
-            catch (Exception) {
-                return BadRequest("Image could not be uploaded.");
+            UploadAndDeleteData uploadAndDeleteResult = await uploadAndDeletePhoto(id, quizBodyData.Files[0]);
+            if (OkResult.Equals(uploadAndDeleteResult.HttpResponseResult, Ok()) )
+            {
+                return uploadAndDeleteResult.HttpResponseResult;
             }
 
-            Quiz updatedQuiz = quizRepository.UpdateQuiz(new Quiz {
+   
+            IActionResult updateQuizResult = updateQuiz(new Quiz
+            {
                 Title = quizBodyData.Title,
-                PictureUrl = fileUrl,
+                PictureUrl = uploadAndDeleteResult.fileName,
                 UserId = currentUserId,
-                Id = id
+                Id = id,
+                LastModified = DateTime.Now
             });
 
-            if (updatedQuiz == null) {
-                return BadRequest("Quiz does not exist.");
-            }
-
-            QuizViewModel quizVm = entityToVmMapper.Map(updatedQuiz);
-            quizVm.PictureUrl = blobService.urlPath.AbsoluteUri.ToString() + "users/" + fileUrl;
-
-            return Ok(new { quizVm.Id, quizVm.Title, quizVm.PictureUrl });
+            return updateQuizResult;
         }
 
         [HttpDelete("{id}")]
         [Authorize]
-        public async Task<IActionResult> Delete(int id) {
+        public async Task<IActionResult> Delete(int id)
+        {
             int currentUserId = User.Claims.GetUserId();
             var deletedQuiz = quizRepository.DeleteQuiz(id, currentUserId);
 
-            if (deletedQuiz != null) {
-                AzureBlobService blobService = new AzureBlobService();
-                await blobService.InitializeBlob();
-                blobService.DeletePhoto(deletedQuiz.PictureUrl);
+            if (deletedQuiz != null)
+            {
+                AzureBlobService blobService = new AzureBlobService(azureContainerName);
+                blobService.DeletePhotoAsync(deletedQuiz.PictureUrl);
                 return Ok();
             }
-            else {
+            else
+            {
                 return BadRequest("Deletion Impossible. Quiz does not exist.");
             }
         }
 
-        private IActionResult FileValidityChecker(NewQuizViewModel quizBodyData) {
+        private IActionResult updateQuiz(Quiz newQuizData)
+        {
+            IActionResult httpResponseResult = Ok();
+            Quiz updatedQuiz = quizRepository.UpdateQuiz(newQuizData);
```

Comment:
update and if null not exist? update means to do action Update and return updated value, Update doesnt have check if exist, separate logic please